src/Cli/Resources/services.xml

```diff
@@ -95,5 +96,45 @@
             <argument>%monitoring.handlers%</argument>
         </service>
 
+        <!-- Challenge Solving -->
+        <service id="challenge_solver.http" class="AcmePhp\Core\Challenge\Http\SimpleHttpSolver">
+            <argument type="service" id="challenge_extractor.http" />
+            <argument type="service" id="output" />
+        </service>
+        <service id="challenge_solver.dns" class="AcmePhp\Core\Challenge\Dns\SimpleDnsSolver">
+            <argument type="service" id="challenge_extractor.dns" />
+            <argument type="service" id="output" />
+        </service>
+
+        <service id="challenge_extractor.dns" class="AcmePhp\Core\Challenge\Dns\DnsDataExtractor">
+            <argument type="service" id="http.base64_encoder" />
+        </service>
+        <service id="challenge_extractor.http" class="AcmePhp\Core\Challenge\Http\HttpDataExtractor" />
```

Comment:
Can you put this line before the DNS extractor (to keep the same order as in the solvers)?